bouncerscript/constants.py

```diff
@@ -1,25 +1,39 @@
-    "firefox-devedition-stub": "^Devedition-\d+\.0b\d+-stub$",
-    "firefox-devedition-latest": "^Devedition-\d+\.0b\d+$",
-    "firefox-devedition-latest-ssl": "^Devedition-\d+\.0b\d+-SSL$",
-    "firefox-beta-stub": "^Firefox-\d+\.0b\d+-stub$",
-    "firefox-beta-latest": "^Firefox-\d+\.0b\d+$",
-    "firefox-beta-latest-ssl": "^Firefox-\d+\.0b\d+-SSL$",
-    "firefox-stub": "^Firefox-\d+\.\d+(\.\d+)?-stub$",
-    "firefox-latest": "^Firefox-\d+\.\d+(\.\d+)?$",
-    "firefox-latest-ssl": "^Firefox-\d+\.\d+(\.\d+)?-SSL$",
-    "firefox-esr-latest": "^Firefox-\d+\.\d+(\.\d+)?esr$",
-    "firefox-esr-latest-ssl": "^Firefox-\d+\.\d+(\.\d+)?esr-SSL$",
-    "firefox-esr-next-latest": "^Firefox-\d+\.\d+(\.\d+)?esr$",
-    "firefox-esr-next-latest-ssl": "^Firefox-\d+\.\d+(\.\d+)?esr-SSL$",
-    "firefox-sha1": "^Firefox-\d+\.\d+(\.\d+)?esr-sha1$",
-    "firefox-sha1-ssl": "^Firefox-\d+\.\d+(\.\d+)?esr-sha1$"
+    'thunderbird-beta-latest': '^Thunderbird-\d+\.0b\d+$',
+    'thunderbird-beta-latest-ssl': '^Thunderbird-\d+\.0b\d+-SSL$',
+    'thunderbird-next-latest': '^Thunderbird-\d+\.\d+(\.\d+)?$',
+    'thunderbird-next-latest-ssl': '^Thunderbird-\d+\.\d+(\.\d+)?-SSL$',
+    'thunderbird-latest': '^Thunderbird-\d+\.\d+(\.\d+)?$',
+    'thunderbird-latest-ssl': '^Thunderbird-\d+\.\d+(\.\d+)?-SSL$',
+    'fennec-beta-latest': '^Fennec-\d+\.0b\d+$',
+    'fennec-latest': '^Fennec-\d+\.\d+(\.\d+)?$',
+    'firefox-devedition-stub': '^Devedition-\d+\.0b\d+-stub$',
+    'firefox-devedition-latest': '^Devedition-\d+\.0b\d+$',
+    'firefox-devedition-latest-ssl': '^Devedition-\d+\.0b\d+-SSL$',
+    'firefox-beta-stub': '^Firefox-\d+\.0b\d+-stub$',
+    'firefox-beta-latest': '^Firefox-\d+\.0b\d+$',
+    'firefox-beta-latest-ssl': '^Firefox-\d+\.0b\d+-SSL$',
+    'firefox-stub': '^Firefox-\d+\.\d+(\.\d+)?-stub$',
+    'firefox-latest': '^Firefox-\d+\.\d+(\.\d+)?$',
+    'firefox-latest-ssl': '^Firefox-\d+\.\d+(\.\d+)?-SSL$',
+    'firefox-esr-latest': '^Firefox-\d+\.\d+(\.\d+)?esr$',
+    'firefox-esr-latest-ssl': '^Firefox-\d+\.\d+(\.\d+)?esr-SSL$',
+    'firefox-esr-next-latest': '^Firefox-\d+\.\d+(\.\d+)?esr$',
+    'firefox-esr-next-latest-ssl': '^Firefox-\d+\.\d+(\.\d+)?esr-SSL$',
+    'firefox-sha1': '^Firefox-\d+\.\d+(\.\d+)?esr-sha1$',
+    'firefox-sha1-ssl': '^Firefox-\d+\.\d+(\.\d+)?esr-sha1$',
+}
+
+PRODUCT_TO_DESTINATIONS_REGEXES = {
+    'fennec': '^(/mobile/releases/.*?/(?:android-api-16|android-x86)/\:lang/fennec-.*\:lang\.(?:android-arm|android-i386)\.apk)$',
+    'firefox': '^(/firefox/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64)/\:lang/(?:firefox|Firefox).*\.(?:bz2|dmg|exe|mar))$',
+    'devedition': '^(/devedition/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64)/\:lang/(?:firefox|Firefox).*\.(?:bz2|dmg|exe|mar))$',
+    'thunderbird': '^(/thunderbird/releases/.*?/(update/)?(?:linux-i686|linux-x86_64|mac|win32|win64)/\:lang/(?:thunderbird|Thunderbird).*\.(?:bz2|dmg|exe|mar))$',
```

Comment:
I'll move them to script_config.json, you're right. It'll be easier in the future. I may move the ALIASES_REGEX too.

Reply:
@JohanLorenzo How do you feel about this? We were chatting in IRC earlier and remembered you had strong feelings against puppet in this case, being concerned about strings getting munged. Do you still feel that way and you think we should keep these regexes here? 

The drawback of this is the one mentioned by Aki, that we'd need to roll-out a new bouncerscript version when/if we add a new platform or so. (quite rare frankly but worth considering).

Reply:
Thank you for the question. Yeah, I feel this is dangerous for 2 reasons : 

1. Like you said, we can mess escaping the regexes in puppet. For the record, the chain is : `developer -> puppet -> json -> python regex`. 
1. The only way to test those regexes becomes end-to-end only. Whereas if the regexes remain in the package, we can easily test them out thanks to unit tests (like you did in the PR :smiley:). 

I understand deploying a new release each time is cumbersome. If you guys think deployment outweighs testing in this case, I'm fine moving them to configs.

Reply:
Ah, I forgot about that `developer -> puppet -> json -> python regex` thing`. Uf, just for that and I'm suddently reluctant too. To be honest, I don't think we're going to see platforms change that often. Additionally, for most of the stuff that we test nowadays, we're usually triggering staging releases so hopefully we're able to catch any potential error there before we move to production and need to do-it in a timely sensitive manner.

I'll let them be here for now.

As to the nighyly bouncer aliases that AKi mentioned, Lisa wanted to tackle that next week. I'll make sure to let her know of this beforehand.

Reply:
1. We can put the regexes directly into `bouncer_scriptworker/templates/script_config.json.erb`, which means it's closer to `developer -> json -> python regex`. If there's another format that works better, e.g. yaml, we could switch script_config.json.erb to script_config.yml.erb. `scriptworker::instance` takes a `$task_script_config` argument that defaults to `${basedir}/script_config.json` but can be any file path. In this example, it would be more `developer -> yaml -> python regex`

   - We can even create a python file in puppet, not a template. As long as we point to and use that file's regexes, it then becomes `developer -> python regex`, although we use puppet to deploy that python regex.

2. For the [now-obsolete] scriptworker docker sha allowlists, we had them in `scriptworker.constants.DEFAULT_CONFIG`, allowing for testing in unit tests. We also overrode them in puppet's config. And whenever we updated the puppet config, we also submitted a PR to update `scriptworker.constants.DEFAULT_CONFIG` to match.

   - downside: this means 2x as many PRs to deploy
   - upside: we can local unittest and end-to-end test
   - upside: when a new docker sha busted the trees, we could deploy the fix quickly via puppet, and were not required to deploy the scriptworker release immediately. The scriptworker patch could ride along with the next scheduled scriptworker release.

Reply:
Since deploying this is Q2 goal and this is not yet blocking as we don't have upcoming new platforms or aliases (or if we do next week when we'll hopefully tackle the nightly bouncer aliases) I'd like to merge this. I moved last comment in https://github.com/mozilla-releng/bouncerscript/issues/30 to continue conversation there and follow-up with a separate PR if we have to.